### PR TITLE
expression: implement vectorized evaluation for `builtinMD5Sig`

### PR DESCRIPTION
**Pull request by mmyj**
### What problem does this PR solve? 
implement vectorized evaluation for builtinMD5Sig, for #12106
### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinEncryptionFunc -run=BenchmarkVectorizedBuiltinEncryptionFunc -args 'builtinMD5Sig'
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinEncryptionFunc/builtinMD5Sig-VecBuiltinFunc-4                     3038            345832 ns/op           51918 B/op       2433 allocs/op
BenchmarkVectorizedBuiltinEncryptionFunc/builtinMD5Sig-NonVecBuiltinFunc-4                  3092            365909 ns/op           51917 B/op       2433 allocs/op
PASS
ok      github.com/pingcap/tidb/expression      2.287s
```
### Check List 
Tests 
- Unit test

### CODE REVIEW

File: expression/builtin_encryption_vec.go [hunk 1]
```diff
@@ -100,11 +103,30 @@ func (b *builtinRandomBytesSig) vecEvalString(input *chunk.Chunk, result *chunk.
...
+		return err
+	}
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		sum := md5.Sum([]byte(str))
```
Comment (XuHuaiyu) [inline expression/builtin_encryption_vec.go]: s/ []byte(str)/ hack.Slice(str)

File: expression/builtin_encryption_vec.go [hunk 2]
```diff
@@ -131,11 +135,30 @@ func (b *builtinRandomBytesSig) vecEvalString(input *chunk.Chunk, result *chunk.
...
+		return err
+	}
+	result.ReserveString(n)
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+		str := buf.GetString(i)
+		sum := md5.Sum(hack.Slice(str))
```
[the remark this answers is not in the document]
Comment (mmyj) [reply expression/builtin_encryption_vec.go]: I will fix it.

[the remark this answers is not in the document]
Comment (mmyj) [reply expression/builtin_encryption_vec.go]: I will fix it.

**Comment by mmyj** [comment]
@SunRunAway PTAL

File: expression/builtin_encryption_vec.go [hunk 3]
```diff
@@ -135,11 +136,35 @@ func (b *builtinRandomBytesSig) vecEvalString(input *chunk.Chunk, result *chunk.
...
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalString(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+	digest := md5.New()
+	for i := 0; i < n; i++ {
+		digest.Reset()
```
Comment (SunRunAway) [inline expression/builtin_encryption_vec.go]: Sorry it's my bad,
we could put `digest.Reset` after `digest.Sum`, so if we've continued before calculating, we need not reset.

Comment (mmyj) [reply expression/builtin_encryption_vec.go]: I will fix it.

**Comment by mmyj** [comment]
@SunRunAway PTAL

**Pull request merged**